guava/src/com/google/common/base/RandomStringUtils.java

```diff
@@ -0,0 +1,179 @@
+
+import java.util.Random;
+
+/**
+ * Provides API for random String generation
+ * @author Petr Havlicek
+ */
+
+public class RandomStringUtils {
+    
+    private static final char ASCII_PRINT_START = 32;
+    private static final char ASCII_PRINT_END = 126;
+    private static final char ASCII_NUMERIC_START = 48;
+    private static final char ASCII_NUMERIC_END = 57;
+    
+    /**
+     * 
+     * @param len length of the returned String
+     * @param chars characters to be used - if chars contains non-alphabetic non-printable characters, they may appear in the returned String
+     * @return String of length len from characters chars
+     */
+    public static String random(int len, char[] chars) {
+        if (chars == null || chars.length == 0) {
+            throw new IllegalArgumentException("The chars array must not be empty");
+        }
+        Random random = new Random();
+        char[] res = new char[len];
+        for (int i = 0; i < len; i++) {
+            res[i] = chars[random.nextInt(chars.length)];
+        }
+        return new String(res);
+    }
+
+    /**
+     * 
+     * @param len length of the returned String
+     * @param chars String od characters to be used - if chars contains non-alphabetic non-printable characters, they may appear in the returned String
+     * @return String of length len from characters chars
+     */
+    public static String random(int len, String chars) {
+        Random random = new Random();
+        char[] res = new char[len];
+        char[] c = new char[chars.length()];
+        chars.getChars(0, c.length, c, 0);
+        for (int i = 0; i < res.length; i++) {
+            res[i] = c[random.nextInt(c.length)];
+        }
+        return new String(res);
+    }
+    
+    /**
+     *
+     * @param len length of the returned String
+     * @param start first usable character
+     * @param end last usable character
+     * @return String of length len from alphanumeric characters (Character.isAlphabetic == true) ranging from start to end
+     */
+    public static String randomAlphanumeric(int len, char start, char end) {
+        if (start > end) {
+            throw new IllegalArgumentException("Start char < end char!");
+        }
+        Random random = new Random();
+        char[] res = new char[len];
+        char gap = (char) (end - start);
+        char ch;
+        for (int i = 0; i < res.length; ) {
+            ch = (char) (random.nextInt(gap) + start);
+            if(Character.isAlphabetic(ch)){
+                res[i] = ch;
+                i++;
+            }
+        }
+        return new String(res);
+    }
+    
+    /**
+     *
+     * @param len length of the returned String
+     * @param start first usable character
+     * @param end last usable character
+     * @return String of length len from numeric characters (Character.isDigit == true) ranging from start to end
+     */
+    public static String randomNumeric(int len, char start, char end) {

```

Comment:
Do we need to mention any documentation about Unicode characters ? For example :

start = '\u2202' and end = '\u2500' - in this case we don't have any numeric characters in this range.